src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo: Todo;
+  setSelectedTodo: (selectedTodo: Todo | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(selectedTodo.userId).then(data => {
+      setUser(data);
+      setIsLoading(false);
+    });
+  }, [selectedTodo.userId]);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    setIsLoading(true);
    
    getUser(selectedTodo.userId).then(data => {
      setUser(data);
    })
    .finally(() => {
      setIsLoading(false);
    });
  }, [selectedTodo.userId]);
```
You should set loader before fetching and remove it in any case after the request, not only in successful